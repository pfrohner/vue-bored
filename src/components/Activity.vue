<template>
  <section class="mb-3">
    <div class="row text-left">
      <div class="col-sm-6">
        <p>You should:</p>
        <label for="name">Activity</label>
        <textarea
          v-model="newActivity.activity"
          class="md-textarea form-control mb-3"
          rows="5"
          id="name"
        />
      </div>
      <div class="col-sm-6">
        <p>Activity details:</p>
        <div class="form-group">
          <label for="type">Type</label>
          <input type="text" class="form-control" id="type" :value="newActivity.type" />
        </div>
        <div class="form-group">
          <label for="participants">Participants</label>
          <input
            type="text"
            class="form-control"
            id="participants"
            :value="newActivity.participants"
          />
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm-6">
        <button
          data-test="add-btn"
          class="btn btn-danger btn-block mb-3"
          @click="$emit('add', newActivity)"
        >Save for later!</button>
      </div>
      <div class="col-sm-6">
        <button
          data-test="get-btn"
          class="btn btn-primary btn-block mb-3"
          @click="$emit('get')"
        >Hit me with a new one!</button>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: "Activity",
  data() {
    return {
      newActivity: {}
    };
  },
  watch: {
    activity: function(attrs) {
      this.newActivity = { ...attrs };
    }
  },
  props: {
    activity: {
      type: Object,
      required: true
    }
  }
};
</script>
