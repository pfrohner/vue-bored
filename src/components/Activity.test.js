import { shallowMount } from '@vue/test-utils'
import Activity from './Activity.vue'

const setup = () => {
  return shallowMount(Activity, {
    propsData: {
      activity: {}
    }
  })
}

describe('Activity', () => {
  it('renders correctly', () => {
    const wrapper = setup()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits the proper event when adding new activity', () => {
    const wrapper = setup()
    const button = wrapper.find('[data-test="add-btn"]')

    button.trigger('click')

    expect(wrapper.emitted().add).toBeTruthy()
  })

  it('emits the appropriate event when getting activity', () => {
    const wrapper = setup()
    const button = wrapper.find('[data-test="get-btn"]')

    button.trigger('click')

    expect(wrapper.emitted().get).toBeTruthy()
  })
})
