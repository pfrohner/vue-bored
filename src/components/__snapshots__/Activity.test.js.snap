// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity renders correctly 1`] = `
<section
  class="mb-3"
>
  <div
    class="row text-left"
  >
    <div
      class="col-sm-6"
    >
      <p>
        You should:
      </p>
       
      <label
        for="name"
      >
        Activity
      </label>
       
      <textarea
        class="md-textarea form-control mb-3"
        id="name"
        rows="5"
      />
    </div>
     
    <div
      class="col-sm-6"
    >
      <p>
        Activity details:
      </p>
       
      <div
        class="form-group"
      >
        <label
          for="type"
        >
          Type
        </label>
         
        <input
          class="form-control"
          id="type"
          type="text"
        />
      </div>
       
      <div
        class="form-group"
      >
        <label
          for="participants"
        >
          Participants
        </label>
         
        <input
          class="form-control"
          id="participants"
          type="text"
        />
      </div>
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col-sm-6"
    >
      <button
        class="btn btn-danger btn-block mb-3"
        data-test="add-btn"
      >
        Save for later!
      </button>
    </div>
     
    <div
      class="col-sm-6"
    >
      <button
        class="btn btn-primary btn-block mb-3"
        data-test="get-btn"
      >
        Hit me with a new one!
      </button>
    </div>
  </div>
</section>
`;
